packages/next/src/server/lib/route-resolver.ts

```diff
@@ -65,14 +76,52 @@ export async function makeResolver(dir: string, nextConfig: NextConfig) {
   const devServer = new DevServer({
     dir,
     conf: nextConfig,
+    hostname: 'localhost',
+    port: 3000,
```

Comment:
Both are necessary for the `devServer.attachRequestMeta(req, parsedUrl)` call to initialize the `__NEXT_INIT_URL` meta for edge invocation.